### PR TITLE
Improve edgecase handling for empty trees

### PR DESCRIPTION
**Pull request by lukasdreyer**
**_Describe your changes here:_**
When trying to run our testsuite on 8 procs, several tests failed, so this PR fixes some of the problems related to empty trees.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: src/t8_forest/t8_forest_partition.cxx [hunk 1]
```diff
@@ -425,14 +425,22 @@ t8_forest_partition_compute_new_offset (t8_forest_t forest)
   SC_CHECK_MPI (mpiret);
 
   if (t8_shmem_array_start_writing (forest->element_offsets)) {
-    t8_gloidx_t *element_offsets = t8_shmem_array_get_gloidx_array_for_writing (forest->element_offsets);
-    for (i = 0; i < mpisize; i++) {
-      /* Calculate the first element index for each process. We convert to doubles to prevent overflow */
-      new_first_element_id = (((double) i * (long double) forest_from->global_num_elements) / (double) mpisize);
-      T8_ASSERT (0 <= new_first_element_id && new_first_element_id < forest_from->global_num_elements);
-      element_offsets[i] = new_first_element_id;
+    if (forest_from->global_num_elements) {
```
Comment (jmark) [inline src/t8_forest/t8_forest_partition.cxx:428]: To increase code readability
```suggestion
    if (forest_from->global_num_elements > 0) {
```

**Review by jmark — APPROVED** [review]

**Comment by jmark** [comment]
@lukasdreyer I ran the test with 3 ranks (also with 8 ranks) in debug mode and it failed. For one and two ranks the following assertion does not fail:

```
mpirun -n 3 fix-empty_tree_handling/test/t8_gtest_iterate_replace
[...]
[t8 0] Building global first descendants for forest 0x13289a0
[t8 1] Building global first descendants for forest 0x25e2320
[t8 2] Building global first descendants for forest 0xa37220
[libsc 0] Abort: Assertion 'last_desc_id < first_desc_id'
[libsc 0] Abort: /home/mark_jo/codes/t8code/main/src/t8_forest/t8_forest_partition.cxx:245
[libsc 0] Abort: Obtained 21 stack frames
[...]
```

See following for full output.
[out.txt](https://github.com/user-attachments/files/16099669/out.txt)

**Comment by lukasdreyer** [comment]
> @lukasdreyer I ran the test with 3 ranks (also with 8 ranks) in debug mode and it failed. For one and two ranks the following assertion does not fail:
> 
> ```
> mpirun -n 3 fix-empty_tree_handling/test/t8_gtest_iterate_replace
> [...]
> [t8 0] Building global first descendants for forest 0x13289a0
> [t8 1] Building global first descendants for forest 0x25e2320
> [t8 2] Building global first descendants for forest 0xa37220
> [libsc 0] Abort: Assertion 'last_desc_id < first_desc_id'
> [libsc 0] Abort: /home/mark_jo/codes/t8code/main/src/t8_forest/t8_forest_partition.cxx:245
> [libsc 0] Abort: Obtained 21 stack frames
> [...]
> ```
> 
> See following for full output. [out.txt](https://github.com/user-attachments/files/16099669/out.txt)

The assertion is definitely wrong and maybe we can talk on the phone how to write a better assertion there. I think during debugging I already once discovered that this assertion is wrong and simply deleted it, but forgot to include it in the PR.

**Comment by lukasdreyer** [comment]
> > @lukasdreyer I ran the test with 3 ranks (also with 8 ranks) in debug mode and it failed. For one and two ranks the following assertion does not fail:
> > ```
> > mpirun -n 3 fix-empty_tree_handling/test/t8_gtest_iterate_replace
> > [...]
> > [t8 0] Building global first descendants for forest 0x13289a0
> > [t8 1] Building global first descendants for forest 0x25e2320
> > [t8 2] Building global first descendants for forest 0xa37220
> > [libsc 0] Abort: Assertion 'last_desc_id < first_desc_id'
> > [libsc 0] Abort: /home/mark_jo/codes/t8code/main/src/t8_forest/t8_forest_partition.cxx:245
> > [libsc 0] Abort: Obtained 21 stack frames
> > [...]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > See following for full output. [out.txt](https://github.com/user-attachments/files/16099669/out.txt)
> 
> The assertion is definitely wrong and maybe we can talk on the phone how to write a better assertion there. I think during debugging I already once discovered that this assertion is wrong and simply deleted it, but forgot to include it in the PR.

I adapted the assertion, so that `t8_gtest_iterate_replace` now should run on multiple procs.
Altough i still think that there can be cases, where this assertion is too strict, but I added a corresponding comment in the commit.

**Pull request merged**